crates/shared/src/order_quoting.rs

```diff
@@ -660,11 +662,32 @@ impl From<&OrderQuoteRequest> for PreOrderData {
 
 impl From<&OrderQuoteRequest> for QuoteParameters {
     fn from(request: &OrderQuoteRequest) -> Self {
+        // The `from` address is currently mandatory for quote requests. However, when
+        // the user did not connect their wallet they still want to get prices
+        // so in that case the field currently gets populated with the 0
+        // address. Using this address in quote verifications will definitely
+        // cause reverts in the simulation so we make an exception here.
+        // TODO: Get rid of this when quotes and price estimates use different
+        // endpoints.
+        let verification = if request.from == H160::zero() {
+            None
+        } else {
+            Some(Verification {
+                from: request.from,
+                receiver: request.receiver.unwrap_or(request.from),
+                sell_token_source: request.sell_token_balance,
+                buy_token_destination: request.buy_token_balance,
+                // TODO get from request
+                pre_interactions: vec![],
+                post_interactions: vec![],
+            })
+        };
```

Comment:
I would remove this bit. Specifically, what I would do instead is in the `/quote` endpoint handler do a `if quote.quality == ::Verified and quote.from.is_zero() { err!() }` instead of having this "0 has special semantics" leak here.

Reply:
> Instead of introducing a new price quality verified (https://github.com/cowprotocol/services/pull/1512) we implicitly verify all quotes where from is set.

I was thinking about not introducing the `Verified` price quality after all (the PR is still not merged as I was not 100% sure how the whole thing would fit together).
If we indeed want to introduce the `Verified` price quality I agree that this special case should be dropped. But I thought it would be nice this way that the frontend wouldn't have to change for now.
LMK what you think about automatic verified quotes vs. new price quality.

Reply:
> LMK what you think about automatic verified quotes vs. new price quality.

Would this would break the existing API? As in, do you want to do a `token.balanceOf(from)` and decide whether or not to verify (which adds some latency)? I.e. what is your condition for deciding to verify quotes?

Reply:
With this PR we would always verify quotes when we can (i.e. we have a non-0 `from`).
This would add latency but hopefully not too much. I can also "feature gate" it behind the `verified` price quality so we don't really change anything immediately.